nix/shell.nix

```diff
@@ -3,8 +3,77 @@
 , ninja
 , gcc-arm-embedded
 , python3
+, python3Packages
+, lib
+, fetchFromGitHub
+, extra-pkgs ? []
 }:
 
+let
+  # Keymap-drawer uses version 3 of `platformdirs` but NixPkgst only has
+  # version 4.2 available. It's easier to brging the package from GitHub than
+  # creating a flake input for an old version of NixPkgs, just to get this one
+  # dependency
+  platformdirs3 = python3Packages.buildPythonPackage rec {
+    name = "platformdirs";
+    version = "3.11.0";
+
+    src = fetchFromGitHub {
+      owner = "platformdirs";
+      repo = "${name}";
+      rev = "${version}";
+      sha256 = "sha256-rMPpxwPbqAtvr3RtKQDisqQnCxnBfZdolMUPpDE+tR4=";
+    };
+
+    format = "pyproject";
+
+    nativeBuildInputs = [
+      python3Packages.hatchling
+      python3Packages.hatch-vcs
+    ];
+
+    meta = {
+      homepage = "https://github.com/platformdirs/platformdirs/tree/3.11.0";
+      description = "A small Python module for determining appropriate platform-specific dirs";
+      license = lib.licenses.mit;
+    };
+  };
+
+  # NixPkgs does not offer `keymap-drawer` as a python package nor a
+  # stand-alone application, so we have  pull the package into our environment
+  keymap-drawer = python3Packages.buildPythonPackage rec {
```

Comment:
I think it would be better to expose this as a package in the flake, and create an output that uses it to build the SVGs.

Reply:
Agree with this one. I'm also thinking that a "local build flake" might be better to tackle the nix shell utility.